src/odin/narrativebuilder.cc

```diff
@@ -1057,14 +1146,32 @@ std::string NarrativeBuilder::FormVerbalTurnInstruction(Maneuver& maneuver,
 
   // Determine which phrase to use
   uint8_t phrase_id = 0;
-  if (!street_names.empty()) {
-    phrase_id = 1;
-  }
-  if (!begin_street_names.empty()) {
-    phrase_id = 2;
-  }
-  if (maneuver.to_stay_on()) {
-    phrase_id = 3;
+  std::string junction_name;
+  std::string guide_sign;
+
+  if (maneuver.HasGuideSign()) {
+    // Set the toward phrase - it takes priority over street names and junction name
```

Comment:
Just noting we may want to revisit the priority ordering of GuideSign > JunctionName > StreetName at some point. Also it may be preferable to announce JunctionName and GuideSign together `Turn left at X toward Y`. No change needed for this PR.

Reply:
I was thinking the same possibility but kept to the current plan for now